backend/test_observer/controllers/test_executions/post_results.py

```diff
@@ -0,0 +1,52 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy import delete
+from sqlalchemy.orm import Session
+
+from test_observer.controllers.test_executions.models import TestResultRequest
+from test_observer.data_access.models import TestCase, TestExecution, TestResult
+from test_observer.data_access.repository import get_or_create
+from test_observer.data_access.setup import get_db
+
+router = APIRouter(tags=["test-results"])
+
+
+@router.post("/{id}/test-results")
+def post_results(
+    id: int,
+    request: list[TestResultRequest],
+    db: Session = Depends(get_db),
+):
+    test_execution = db.get(TestExecution, id)
+
+    if test_execution is None:
+        raise HTTPException(status_code=404, detail="TestExecution not found")
+
+    for result in request:
+        test_case = get_or_create(
+            db,
+            TestCase,
+            filter_kwargs={"name": result.name},
+            creation_kwargs={
+                "category": result.category,
+                "template_id": result.template_id,
+            },
+        )
+
+        db.execute(
+            delete(TestResult).where(
+                TestResult.test_execution_id == test_execution.id,
+                TestResult.test_case_id == test_case.id,
+            )
+        )
+
+        test_result = TestResult(
+            test_execution=test_execution,
+            test_case=test_case,
+            status=result.status,
```

Comment:
I wonder if the test_execution.status should be recomputed based on the logic in https://github.com/canonical/test_observer/pull/236/files. Or maybe it would make sense that the test_execution.status is just a computed value.

Reply:
I'm imagining a workflow in which users of TO would do the following:
1. Hit the start test endpoint
2. Submit results through this post endpoint (perhaps multiple times too)
3. End the test execution by sending a patch request with the `COMPLETED` status
In such a workflow if a user hits the endpoint they didn't necessarily finish yet. So perhaps it's best not to change the status of the test execution in case they want to submit more results (at least until they send COMPLETED)

Reply:
Technically only the `PASSED` and `FAILED` statuses could be computed values. But I don't see a benefit of making them so